@keyv/compress-gzip and @keyv/compress-brotli will not build with Typescript

When using either of @keyv/compress-gzip and @keyv/compress-brotli in a Typescript project, it will not build and throw the following errors:

```
node_modules/@keyv/compress-gzip/src/index.d.ts:3:14 - error TS7006: Parameter 'options' implicitly has an 'any' type.

3  constructor(options?);
               ~~~~~~~~

node_modules/@keyv/compress-gzip/src/index.d.ts:4:2 - error TS1040: 'async' modifier cannot be used in an ambient context.

4  async compress(value: any);
   ~~~~~

node_modules/@keyv/compress-gzip/src/index.d.ts:4:8 - error TS7010: 'compress', which lacks return-type annotation, implicitly has an 'any' return type.

4  async compress(value: any);
         ~~~~~~~~
node_modules/@keyv/compress-gzip/src/index.d.ts:5:8 - error TS7010: 'decompress', which lacks return-type annotation, implicitly has an 'any' return type.

5  async decompress(value: any);
         ~~~~~~~~~~

node_modules/@keyv/compress-gzip/src/index.d.ts:6:2 - error TS1040: 'async' modifier cannot be used in an ambient context.

6  async serialize(value: any);
   ~~~~~

node_modules/@keyv/compress-gzip/src/index.d.ts:6:8 - error TS7010: 'serialize', which lacks return-type annotation, implicitly has an 'any' return type.

6  async serialize(value: any);
         ~~~~~~~~~

node_modules/@keyv/compress-gzip/src/index.d.ts:7:2 - error TS1040: 'async' modifier cannot be used in an ambient context.

7  async deserialize(value: any);
   ~~~~~

node_modules/@keyv/compress-gzip/src/index.d.ts:7:8 - error TS7010: 'deserialize', which lacks return-type annotation, implicitly has an 'any' return type.

7  async deserialize(value: any);
         ~~~~~~~~~~~
```

The same errors are returned when using @keyv/compress-brotli.

Apparently, in index.d.ts, several variables lack types and the async function cannot be used as is.

--- a/packages/compress-brotli/src/index.d.ts
+++ b/packages/compress-brotli/src/index.d.ts
@@ -4,10 +4,10 @@ import type Brotli, {CompressResult} from 'compress-brotli';
 declare class KeyvBrotli {
 	brotli: Brotli;
 	constructor(options?: KeyvBrotli.Options);
-	async compress(value: InputType | number | boolean, options?: BrotliOptions);
-	async decompress(value: InputType | number | boolean, options?: BrotliOptions);
-	async serialize(value: any);
-	async deserialize(value: any);
+	compress(value: InputType | number | boolean, options?: BrotliOptions): Promise<any>;
+	decompress(value: InputType | number | boolean, options?: BrotliOptions): Promise<any>;
+	serialize(value: any): Promise<any>;
+	deserialize(value: any): Promise<any>;
 }
 
 declare namespace KeyvBrotli {

--- a/packages/compress-brotli/test/test.js
+++ b/packages/compress-brotli/test/test.js
@@ -12,14 +12,6 @@ const {keyvCompresstionTests} = require('@keyv/test-suite');
 
 keyvCompresstionTests(test, new KeyvBrotli());
 
-test('number array compression/decompression', async t => {
-	const keyv = new KeyvBrotli();
-	const array = [4, 5, 6, 7];
-	const compressed = await keyv.compress(array);
-	const decompressed = await keyv.decompress(compressed, {});
-	t.deepEqual(decompressed, array);
-});
-
 test('object type compression/decompression', async t => {
 	const keyv = new KeyvBrotli();
 	const object = {

--- a/packages/compress-gzip/package.json
+++ b/packages/compress-gzip/package.json
@@ -47,6 +47,7 @@
   },
   "homepage": "https://github.com/jaredwray/keyv",
   "dependencies": {
+    "@types/pako": "^2.0.0",
     "json-buffer": "^3.0.1",
     "pako": "^2.0.4"
   },

--- a/packages/compress-gzip/src/index.d.ts
+++ b/packages/compress-gzip/src/index.d.ts
@@ -1,10 +1,13 @@
+import pako from 'pako';
+
 declare class KeyvGzip {
 	opts: any;
-	constructor(options?);
-	async compress(value: any);
-	async decompress(value: any);
-	async serialize(value: any);
-	async deserialize(value: any);
+	constructor(options?: pako.DeflateFunctionOptions | pako.InflateFunctionOptions);
+	compress(value: pako.Data | string, options?: pako.DeflateFunctionOptions): Promise<Uint8Array>;
+	decompress(value: pako.Data, options?: pako.InflateFunctionOptions & {to: 'string'}): Promise<string>;
+	decompress(value: pako.Data, options?: pako.InflateFunctionOptions): Promise<Uint8Array>;
+	serialize(value: any): Promise<any>;
+	deserialize(value: any): Promise<any>;
 }
 
 export = KeyvGzip;

--- a/packages/compress-gzip/test/test.js
+++ b/packages/compress-gzip/test/test.js
@@ -4,7 +4,7 @@ const KeyvGzip = require('this');
 
 keyvCompresstionTests(test, new KeyvGzip());
 
-test('number array compression/decompression', async t => {
+test('number array compression/decompression with Unit8Array', async t => {
 	const keyv = new KeyvGzip();
 	const array = new Uint8Array([4, 5, 6, 7]);
 	const compressed = await keyv.compress(array);

--- a/packages/test-suite/src/compression.js
+++ b/packages/test-suite/src/compression.js
@@ -10,6 +10,13 @@ const keyvCompressionTests = (test, compression) => {
 		await keyv.clear();
 	});
 
+	test('number array compression/decompression', async t => {
+		const array = JSON.stringify([4, 5, 6, 7]);
+		const compressed = await compression.compress(array);
+		const decompressed = JSON.parse(await compression.decompress(compressed));
+		t.deepEqual(decompressed, array);
+	});
+
 	test.serial('compression/decompression using default options', async t => {
 		const compressed = await compression.compress('whatever');
 		t.not(compressed, 'whatever');

--- a/packages/test-suite/src/compression.js
+++ b/packages/test-suite/src/compression.js
@@ -14,7 +14,7 @@ const keyvCompressionTests = (test, compression) => {
 		const array = JSON.stringify([4, 5, 6, 7]);
 		const compressed = await compression.compress(array);
 		const decompressed = JSON.parse(await compression.decompress(compressed));
-		t.deepEqual(decompressed, array);
+		t.is(decompressed, array);
 	});
 
 	test.serial('compression/decompression using default options', async t => {

--- a/packages/test-suite/src/compression.js
+++ b/packages/test-suite/src/compression.js
@@ -14,7 +14,7 @@ const keyvCompressionTests = (test, compression) => {
 		const array = JSON.stringify([4, 5, 6, 7]);
 		const compressed = await compression.compress(array);
 		const decompressed = JSON.parse(await compression.decompress(compressed));
-		t.is(decompressed, array);
+		t.deepEqual(decompressed, [4, 5, 6, 7]);
 	});
 
 	test.serial('compression/decompression using default options', async t => {

--- a/packages/test-suite/src/compression.js
+++ b/packages/test-suite/src/compression.js
@@ -17,6 +17,14 @@ const keyvCompressionTests = (test, compression) => {
 		t.is(decompressed, 'whatever');
 	});
 
+	test.serial('compression/decompression with number', async t => {
+		const number_ = JSON.stringify(5);
+		const compressed = await compression.compress(number_);
+		t.not(compressed, 5);
+		const decompressed = JSON.parse(await compression.decompress(compressed));
+		t.is(decompressed, 5);
+	});
+
 	// Test serialize compression
 	test('serialize compression', async t => {
 		const json = await compression.serialize({value: 'whatever'});

